.github/workflows/postman-metrics.yml

```diff
@@ -0,0 +1,51 @@
+name: Compare and Send Postman Fork Data
+
+on:
+  workflow_dispatch:
+  workflow_call:
+    secrets: # all secrets are passed explicitly in this workflow
+      SLACK_WEBHOOK_URL:
+        required: true
+
+jobs:
+  compare-and-send-forks:
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Fetch previous week's data from cache
+      uses: actions/cache/restore@v4
+      with:
+        path: ./tools/postman/openapi/fork-data.json
+        key: fork-data-${{ github.run_id }}
+        restore-keys: |
+          fork-data-
+
+    - name: Save previous week's data to file
+      working-directory: ./tools/postman/openapi
+      run: |
+        if [ -f fork-data.json ]; then
+          cp fork-data.json previous-fork-data.json
+        else
```

Comment:
I don't understand why you are still creating the file if you don't have data. Could you clarify a bit more here? Can't you just don't compare if there are no data?

Reply:
My thought process was that if there is no previous data then it is equivalent to if there was no collections the previous week, which was an edge case I was testing.

I agree that it does look confusing so I will add a check that the file exists in the bash script rather than creating an empty result

Reply:
Added check in bash script